perceval/backend.py

```diff
@@ -383,12 +383,15 @@ def _remove_key_from_nested_dict(nested_dict, path_to_field):
     if len(path_to_field) == 0:
         return
 
-    key = path_to_field[0]
-
-    if len(path_to_field) == 1:
-        nested_dict.pop(key)
+    if isinstance(nested_dict, list):
+        for dict_item in nested_dict:
+            _remove_key_from_nested_dict(dict_item, path_to_field)
     else:
-        _remove_key_from_nested_dict(nested_dict[key], path_to_field[1:])
+        key = path_to_field[0]
+        if len(path_to_field) == 1:
+            nested_dict.pop(key)
+        else:
+            _remove_key_from_nested_dict(nested_dict[key], path_to_field[1:])
```

Comment:
The name `nested_dict` for something that might not be a `dict` is misleading. Do you mind to rename it to `nested_dictlist`. The same with the name of the function.

You can also rename `dict_item` to `item`.